src/main.rs

```diff
@@ -26,16 +48,129 @@ fn main() -> anyhow::Result<()> {
     let t1 = cpu_times_percpu()?;
     thread::sleep(time::Duration::from_millis(delay));
     let t2 = cpu_times_percpu()?;
-    let result: String = t2
+    let (ratios, result) = t2
         .iter()
         .zip(t1.iter())
         .map(|(a, b)| a - b)
         .map(|c| (c.total().as_secs_f64() - c.idle().as_secs_f64()) / c.total().as_secs_f64())
-        .map(|v| (v * 100.0) as usize)
-        .map(|v| v / 12)
-        .map(|i| bars[i])
-        .collect();
+        .map(|v| (v, (v * 100.0) as usize))
+        .map(|(r, v)| (r, v / 12))
+        .map(|(r, i)| (r, bars[i]))
+        .unzip::<_, _, Vec<_>, String>();
+
+    let include_color = m.occurrences_of("color") == 1;
+    let use_tmux_color = m.occurrences_of("tmux-color") == 1;
+
+    if include_color && use_tmux_color {
+        eprintln!("You cannot set both -c/--color and -C/--tmux-color");
+        std::process::exit(1);
+    }
+
+    if include_color {
+        print_with_colors(result, ratios, ColorMode::Ansi);
+    } else if use_tmux_color {
+        print_with_colors(result, ratios, ColorMode::Tmux);
+    } else {
+        print!("{}", result);
+    }
 
-    println!("{}", result);
     Ok(())
 }
+
+const GREEN_UNTIL: f64 = 2.0;
+const YELLOW_UNTIL: f64 = 8.0;
```

Comment:
Can you help me understand the constants here? is this tailored to a 8-core machine?

Reply:
Good question. I don't quite understand what the numbers mean so I just looked at the values and tweaked them until it seems right. But yeah I guess they might assume you're on an 8-core machine, which I am.

Do you know of a more general way to do this? Can the load numbers be converted to a ratio between 0 and 1 somehow?

Reply:
you probably want to use [the loadavg() function](https://docs.rs/psutil/3.0.1/psutil/host/fn.loadavg.html) that is going to give you the load average numbers like you get on `uptime`. Rule of thumb is that the number is ok as long as it is below `number of cores`, so a loadavg of `8.0` on a 8-core machine is supposedly fine (you have cpu resources being fully utilized without any contention).

The "fun" part is figuring out a yellow number that would make sense 🙃 

my suggestion - we take loadavg from the last minute - this should serve our purpose to see if the machine is swamped doing something on our tmux status bars:

for coloring, how about this:

```
0-70% =>green
70-100% => yellow
100% and beyond => red
```

This is all very much subjective, so what do you think?

Reply:
Sounds good. I'll try to implement it and report back 😊